### PR TITLE
APP-951-Claims status card

### PR DESCRIPTION
**Pull request by tarikstafford**
## [APP-951]

- added claims card to home

## Checklist

- [ ] Dark/light mode verification
- [ ] Landscape verification
- [ ] iPad verification
- [ ] iPod/iPhone 12 mini/iPhone SE verification

[APP-951]: https://hedvig.atlassian.net/browse/APP-951?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Projects/Home/Sources/ClaimStatus.swift [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            hPillOutline(text: L10n.Home.ClaimCard.Pill.claim.uppercased())
+        } else if self == .reopened {
+            hPillFill(text: L10n.Home.ClaimCard.Pill.reopened.uppercased(), backgroundColor: hTintColor.orangeTwo)
+        } else {
+            EmptyView()
+        }
+    }
+}
+
+#if DEBUG
```
Comment (sampettersson) [inline Projects/Home/Sources/ClaimStatus.swift:105]: I think we should avoid DEBUG statements, they don't build on CI which means that we might get build errors locally which we won't get in CI 😅

File: Projects/Home/Sources/ClaimStatusBar.swift [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    let status: Claim.ClaimStatus
+    @State var currentStatus: Claim.ClaimStatus
+
+    @hColorBuilder var barColor: some hColor {
+        switch (status, currentStatus) {
+        case (.reopened, .reopened):
+            hTintColor.orangeOne
+        case (_, .closed):
+            hTintColor.lavenderOne
+        case (let x, let y) where x == y:
```
Comment (sampettersson) [inline Projects/Home/Sources/ClaimStatusBar.swift:16]: What does this mean haha

File: Projects/hCore/Sources/Previews.swift [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import SwiftUI
+
+struct ScreenPreview<Screen: View>: View {
+    var screen: Screen
+
+    let colorScheme: ColorScheme
+
+    var body: some View {
+        ForEach(devices) { device in
+            NavigationView {
```
Comment (sampettersson) [inline Projects/hCore/Sources/Previews.swift:10]: Maybe we should use the JourneyPreviewer here to make sure the previews behave exactly as they would in production 🤔

File: Projects/Home/Sources/ClaimStatus.swift [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    var body: some View {
+        VStack(alignment: .leading, spacing: 0) {
+            HStack(alignment: .top) {
+                ClaimPills(claim: claim)
+                Spacer()
+                Image(uiImage: hCoreUIAssets.chevronRight.image).tint(hLabelColor.secondary)
+            }
+            .padding([.leading, .trailing], 10)
+            Spacer().frame(height: 23)
+            hText("New insurance case")
```
Comment (sampettersson) [inline Projects/Home/Sources/ClaimStatus.swift:18]: Missing textkey

**Review by sampettersson — CHANGES_REQUESTED** [review]

File: Projects/Home/Sources/ClaimStatus.swift (hunk 1 above)
Comment (tarikstafford) [reply Projects/Home/Sources/ClaimStatus.swift:105]: I figured we wouldn't want them in prod but I guess it doesn't matter.

File: Projects/Home/Sources/ClaimStatusBar.swift (hunk 2 above)
Comment (tarikstafford) [reply Projects/Home/Sources/ClaimStatusBar.swift:16]: a case where the two values are equal

File: Projects/App/Sources/Journeys/ClaimsJourney.swift [hunk 5]
```diff
@@ -13,9 +13,19 @@ extension AppJourney {
                 ClaimsAskForPushnotifications(),
                 style: .detented(.large, modally: false)
             ) { _ in
-                AppJourney
-                    .claimsChat()
-                    .withJourneyDismissButton
+                AppJourney.embark(Embark(name: name), storeOffer: false) { result in
```
Comment (sampettersson) [inline Projects/App/Sources/Journeys/ClaimsJourney.swift:16]: I guess we should do a non-offer embark function?

File: Projects/Home/Sources/ClaimStatusBar.swift (hunk 2 above)
Comment (sampettersson) [reply Projects/Home/Sources/ClaimStatusBar.swift:16]: Call it lhs, rhs, maybe instead? its a bit clearer, x and y makes me think about positioning 😂

Comment (tarikstafford) [reply Projects/Home/Sources/ClaimStatusBar.swift:16]: haha oh true.

File: Projects/Home/Sources/ClaimSectionLoading.swift [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+            getter: { $0.claims ?? [] },
+            fetchAction: .fetchClaims,
+            shouldPoll: $shouldPoll
+        ) { claims in
+            claimsSection(claims)
+        }
+    }
+}
+
+struct Poller<S: Store, Value: Equatable, Content: View>: View {
```
Comment (sampettersson) [inline Projects/Home/Sources/ClaimSectionLoading.swift:36]: Move this to hCore maybe?

File: Projects/Home/Sources/ClaimStatus.swift [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+struct ClaimStatus: View {
+    @State var claim: Claim
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: 0) {
+            HStack(alignment: .top) {
+                ClaimPills(claim: claim)
+                Spacer()
+                //Image(uiImage: hCoreUIAssets.chevronRight.image).tint(hLabelColor.secondary)
```
Comment (sampettersson) [inline Projects/Home/Sources/ClaimStatus.swift:14]: Remove this.

File: Projects/hCore/Sources/UIViewController+LifeCycleProxy.swift [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import Flow
+import UIKit
+
+public class LifeCycleProxyViewController: UIViewController {
```
Comment (sampettersson) [inline Projects/hCore/Sources/UIViewController+LifeCycleProxy.swift:4]: What is this for? :o

**Review by sampettersson — CHANGES_REQUESTED** [review]

**Pull request merged**